File: docs/guides/python/setHeaderUserIDThenSaveObjects.py
from algoliasearch.search.client import SearchClientSync
from json import loads


playlists = []  # Your records

_client = SearchClientSync("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")


for playlist in playlists:
    try:
        playlist_user_id = playlist["userID"]
        _client.save_objects(
            index_name="<YOUR_INDEX_NAME>",
            objects=playlists,
            wait_for_tasks=False,
            batch_size=1000,
            request_options={
                "headers": loads("""{"X-Algolia-User-ID":playlistUserID}"""),
            },
        )
    except Exception as e:
        print(f"Error: {e}")

File: docs/guides/go/src/setHeaderUserIDThenSaveObjects.go
package main

import (
	"github.com/algolia/algoliasearch-client-go/v4/algolia/search"
)

func setHeaderUserIDThenSaveObjects() {
	playlists := []map[string]any{{ /* Your records */ }}

	client, err := search.NewClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")
	if err != nil {
		// The client can fail to initialize if you pass an invalid parameter.
		panic(err)
	}

	for _, playlist := range playlists {
		playlistUserID := playlist["userID"]

		_, err := client.SaveObjects(
			"<YOUR_INDEX_NAME>",
			playlists,
			search.WithWaitForTasks(false),
			search.WithBatchSize(1000),
			search.WithHeaderParam("X-Algolia-User-ID", playlistUserID),
		)
		if err != nil {
			panic(err)
		}
	}
}
